### ISSUE TITLE
Cannot access /cms

### ISSUE THREAD

**Issue by bjorntheart** [themosis/framework#156]
Somehow I cannot access my `/cms` from the URL. I get a page saying:
> No route defined for this request.. with a **Back Home** button.

I even tried accessing it via /cms/wp-admin but get the same error.

**Comment by bjorntheart** [comment]
Earlier today I installed the bookstore example and got it working in the sense that I could access the `/cms`. I understand that this is not a working example, but the point is I could access my `/cms` like I could access my `/cms` on another install of themosis. Later today I tried to access the bookstore example again and just get a blank page. Same as with my other Themosis install.
I later checked in the `database:wp_options` and noticed my `home` and `siteurl` was exactly the same. According to the config in the `.env.php` files the siteurl should have `/cms` appended to it. 
Why would the values in the DB then just change?
I wish I had more debug information for you @jlambe to help troubleshoot the problem. I just cannot for the life of me figure out why my themosis installs stop working

**Comment by jlambe** [comment]
I've also noticed the issue in the database. I suppose this is a WordPress Core issue which doesn't check correctly the defined constants. I need to check the WordPress track to see if this issue is listed. This is also explained in the multisite documentation as this has to be fixed in order to work properly with the multisite feature.

**Comment by bjorntheart** [comment]
Do think this is why my sites stop working?

**Comment by jlambe** [comment]
What do you mean by "stop working"?
- Get a home button on undefined request for `/cms`
- Blank page on everything and can't access the admin? (note that the theme do not allow access to cms/wp-admin/ on non-logged users).
- Anything else?

Probably will have to not authorize access to `cms/` URI but I'll need some server aficionado for help to properly configure this. Also, haven't verified if WordPress prevent page/post creation on `cms` uri.

**Comment by bjorntheart** [comment]
Ok, let me try again.

I've installed a Themosis and running it off of `http://themosis.dev`. After the install I could access `http://themosis.dev/cms/` which is the WP backend right? No problem! When I browsed to my site again later that day and trying it again the next day or so, I couldn't access the WP backend on the `/cms` URL.
This is strange, yes I know, but I've encountered it with other fresh installs of the framework too. 
Maybe I am setting something in the WP backend that does something unexpectedly. 

I'll install a fresh copy and test and report back.

**Comment by jlambe** [comment]
The `cms/` URI is the whole WordPress. The admin is located at `cms/wp-admin` URI actually. Currently all my themosis instances are sending a 404 error for the `cms/` URI which is catch by the `Route::any('404')` and if not set, the default "No route defined for this request" message should show.

But the rest of your project is working or you get a blank screen everywhere?

**Comment by bjorntheart** [comment]
Ah I see. Wouldn't it make sense to redirect to the admin when hitting `/cms`, since we are not using the other parts of WP directly but rather through the framework?

I think this was the issue. Haven't got any blank pages yet, but will keep testing Julien :+1:

**Comment by jlambe** [comment]
Yes and no, don't want to redirect unwanted users to the admin part even if they have to be logged-in to get there. Probably we have to think about this a little more :+1:

**Comment by bjorntheart** [comment]
Understood :-) One can just redirect the user back to the home page when accessing the `/cms`, but it still won't stop them from guessing or accessing the admin login. Given the nature of CMS, an admin backend to edit content is always there. Anyone can access the admin URL but does not have permissions to log in. I really don't know how one can 'hide' the admin section from unwanted users.

Maybe I am misinterpreting what your saying :P

**Comment by nwrightau** [comment]
I'm having this issue as well, looking in the database the site url & home url are the same - changing them back to what they should be gives me a no route defined error at /cms and redirects me to the homepage with /cms/wp-admin.

As above it only seems to happen after a while but I really need to access the CMS again. It's a mutl-site install, how do I gain access?

**Comment by jlambe** [comment]
The URL to log in is `cms/wp-login.php` or simply use `login` if permalink is active. There is nothing at the `cms` URI. Also note that the theme prevent people to access the `cms/wp-admin` URI if you're not logged in. Use the `login` URL instead.

Look at the theme `application.config.php` access property which defines a list of people who can access the wp-admin URI if logged in.

**Comment by nwrightau** [comment]
Thankyou! I found it in the end, perhaps this should be in an FAQ section? Would be happy to write a few things and submit them as someone new to using themosis.

**Comment by jlambe** [comment]
A FAQ or probably a forum where people will be able to help each other. In a forum, I'll be able to defined a FAQ or commonly asked issues thread.
